pytest_mpl/kernels.py

```diff
@@ -0,0 +1,230 @@
+"""
+This module contains the supported hashing kernel implementations.
+
+"""
+import hashlib
+from abc import ABC, abstractmethod
+
+import imagehash
+from PIL import Image
+
+#: The default hamming distance bit tolerance for "similar" imagehash hashes.
+DEFAULT_HAMMING_TOLERANCE = 4
+
+#: The default imagehash hash size (N), resulting in a hash of N**2 bits.
+DEFAULT_HASH_SIZE = 16
+
+#: Level of image detail (high) or structure (low) represented by phash .
+DEFAULT_HIGH_FREQUENCY_FACTOR = 4
```

Comment:
I'd hope they're reasonable defaults, but it's difficult to judge. They're the defaults we use on SciTools/iris, however I did bump the tolerance from 2 to 4 to get the phash tests to pass on a Red hat Linux distro, rather than Ubuntu.

Typically, I'd be really nervous about raising the default tolerance higher than 4, as it might hide a genuine change that requires manual inspection; given that we're attempting to test images that should pretty much remain invariant.

Reply:
Yeah, it's difficult to understand what effect each of the defaults will have. Not sure how useful this is, but [here's some code](https://gist.github.com/ConorMacBride/a1d8d3239b462bcce3b2fde9b1f24a60) to see what the image looks like during the phash algorithm. A lot of the detail is removed but changes should be detected by subtle differences in the greyscale value. A thought: maybe we should increase the default hash size and the default hamming tolerance so people will have more flexibility when choosing the tolerance? Although if the current values work for SciTools then it's probably safer to use those as a default.

![phash](https://user-images.githubusercontent.com/25007832/170870105-29fa0d9e-40cb-499f-b566-148967e6957d.png)

Reply:
I guess we can bank what we've got here, as it seems like a reasonable starting point.

Then let the dust settle on this PR, and it'll give you guys time to play and experiment. If there's a concensus, then tweak the defaults based on your experience of how phash behaves.

Does that sound reasonable?